source/io/module_javaScriptInvoke.js

```diff
@@ -303,21 +290,47 @@ var assignJavaScriptInvoke;
             REJECTED: 'REJECTED'
         };
 
-        var updateReturnValue = function (functionName, returnValueRef, returnUserValue, errorValueRef) {
-            if (returnValueRef === undefined) {
-                // User doesn't want to return value. We get here, if we're passing '*' for return parameter in VIA code
+        var hasExecutionError = function (returnValue) {
+            return returnValue instanceof Error;
+        };
+
+        var reportExecutionError = function (functionName, returnValue, errorValueRef, completionCallbackStatus, isInternalFunction) {
+            if (!hasExecutionError(returnValue)) {
```

Comment:
Yea it is redundant. I actually removed it once because I realized I couldn't cover it in tests, but then added it back to be defensive to changes in the future. I don't feel strongly either way and can remove it if that's preferred, but otherwise leaving it.